enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,450 @@
+- OLM support for the user to specify node affinity overrides when deploying an operator and ecosystem improvements.
+- Improve console's content filtering by making it architecture aware and passing that information to OLM. This will include console
+  knowing about the architecture of different nodes and showing catalog content only relevant to those architectures and also
+  providing a way for users to override node affinity settings in an operator.
+- oc debug (and other commands) support for compute architecture.
+- support toolbox.
+- support for Power and Z architectures.
+- Hypershift support for heterogeneous management+worker clusters.
+- Upgrade support in CVO/Cincinnati for heterogeneous release payloads through a heterogeneous graph. Migration of a homogeneous cluster
+would be explicitly done through the upgrade command  by specifying the `--to-image` flag.
+
+Phase 3: Quality of life improvements
+- First class install support - ability to create non-control-plane-architecture workers at install time and generate machineconfig
+and machineset manifests for the same.
+- Possibly reduce mirroring costs by looking into things like sparse manifestlist.
+- CVO migration support for a homogeneous cluster to a heterogeneous payload
+
+### User Stories
+
+- As an enterprise customer looking to save compute resource cost, I want to be able to run part of my application workload on ARM instead of x86.
+- As the owner of an application that runs partially on x86 and partially on ARM, I want to be able to run the entire application within a
+single cluster so I do not need to pay the cost of higher latency between clusters, or worry about the HA-resilience of cross-cluster bridges.
+
+### Workflow Description
+
+A heterogeneous cluster installation will have the same experience as installing a homogeneous cluster. This is the long term vision. However in
+the short term, for the 4.11 Tech Preview, some additional steps will be necessary as IPI installs are not supported yet. The flow will be the
+following:
+- Pick a heterogeneous payload to install from https://multi.ocp.releases.ci.openshift.org/
+- override the release image before creating a cluster by setting this env variable - OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE
+- install a homogeneous x86 cluster as you normally would on AWS/Azure
+- add machinesets of ARM architecture following [these][creating-machinesets] instructions. Make sure to match the instance types and the
+corresponding bootimages based on the architecture of the instance.
+
+### API Extensions
+
+- Imagestreams API will undergo changes described in this [manifestlist enhancement section][imagestreams-api-changes]
+- Upgrades to a heterogeneous release payload will require an additional architecture field in the ClusterVersion API's spec which indicates the
+type of cluster, which CVO can use to fetch the corresponding upgrade graph.
```

Comment:
today there is no architecture property in the cluster spec. CVO figures out the architecture based on GOARCH and selects the appropriate graph. Going forward, CVO will also look at the release image metadata which in case of heterogeneous has the arch property "multi" and make a decision based on this. some of the design details are captured here: https://issues.redhat.com/browse/OTA-658. i will update with the latest discussion.